GitOut/Features/Git/LocalGitRepository.cs

```diff
@@ -94,8 +98,98 @@ await foreach (string line in head.ReadLines())
+            await foreach (string line in status.ReadLinesAsync())
+            {
+                string[] statusLines = line.Split(new char[] { '\0' }, StringSplitOptions.RemoveEmptyEntries);
+                for (int i = 0; i < statusLines.Length; ++i)
+                {
+                    string? statusChange = statusLines[i];
+                    IGitStatusChangeBuilder builder = GitStatusChange.Parse(statusChange);
+                    if (builder.Type == GitStatusChangeType.RenamedOrCopied)
+                    {
+                        builder.MergedFrom(statusLines[++i]);
+                    }
+                    statusChanges.Add(builder.Build());
+                }
+            }
+            CachedStatus = new GitStatusResult(statusChanges);
+            return CachedStatus;
+        }
+
+        public async Task<GitDiffResult> ExecuteDiffAsync(GitStatusChange change, DiffOptions options)
+        {
+            var argumentBuilder = new StringBuilder("--no-optional-locks diff --no-color ");
+            if (options.Cached)
+            {
+                argumentBuilder.Append("--cached ");
+            }
+            if (options.IgnoreAllSpace)
+            {
+                argumentBuilder.Append("--ignore-all-space ");
+            }
+            argumentBuilder.Append($"-- {change.Path}");
+            var args = GitProcessOptions.FromArguments(argumentBuilder.ToString());
+
+            IGitProcess diff = CreateProcess(args);
+            IGitDiffBuilder builder = GitDiffResult.ResultFor(change, options);
+            await foreach (string line in diff.ReadLinesAsync())
+            {
+                builder.Feed(line);
+            }
+            return builder.Build();
+        }
+
+        public Task ExecuteAddAllAsync()
+        {
+            IGitProcess status = CreateProcess(GitProcessOptions.FromArguments("add --all"));
+            return status.ExecuteAsync();
+        }
+
+        public Task ExecuteResetAllAsync()
+        {
+            IGitProcess status = CreateProcess(GitProcessOptions.FromArguments("reset HEAD"));
+            return status.ExecuteAsync();
+        }
+
+        public Task ExecuteAddAsync(GitStatusChange change)
+        {
+            IGitProcess status = CreateProcess(GitProcessOptions.FromArguments($"add {change.Path}"));
+            return status.ExecuteAsync();
+        }
+
+        public Task ExecuteResetAsync(GitStatusChange change)
+        {
+            IGitProcess status = CreateProcess(GitProcessOptions.FromArguments($"reset -- {change.Path}"));
+            return status.ExecuteAsync();
+        }
+
+        public Task ExecuteCommitAsync(string message)
+        {
+            IGitProcess status = CreateProcess(GitProcessOptions.FromArguments($"commit -m \"{message.Replace("\"", "\\\"")}\""));
+            return status.ExecuteAsync();
+        }
```

Comment:
these can be one lined
```suggestion
        public Task ExecuteAddAllAsync()
            => CreateProcess(GitProcessOptions.FromArguments("add --all")).ExecuteAsync();

        public Task ExecuteResetAllAsync()
            => CreateProcess(GitProcessOptions.FromArguments("reset HEAD")).ExecuteAsync();

        public Task ExecuteAddAsync(GitStatusChange change)
            => CreateProcess(GitProcessOptions.FromArguments($"add {change.Path}")).ExecuteAsync();

        public Task ExecuteResetAsync(GitStatusChange change)
            => CreateProcess(GitProcessOptions.FromArguments($"reset -- {change.Path}")).ExecuteAsync();

        public Task ExecuteCommitAsync(string message)
            => CreateProcess(GitProcessOptions.FromArguments($"commit -m \"{message.Replace("\"", "\\\"")}\"")).ExecuteAsync();
```